lib/msfl_visitors/renderers/chewy_renderer.rb

```diff
@@ -0,0 +1,41 @@
+module MSFLVisitors
+  module Renderers
+    class ChewyRenderer
+      BINARY_OPERATORS = {
+        Nodes::And              => ' & ',
+        Nodes::GreaterThan      => ' > ',
+        Nodes::LessThan         => ' < ',
+        Nodes::GreaterThanEqual => ' >= ',
+        Nodes::LessThanEqual    => ' <= ',
+        Nodes::Equal            => ' == ',
+      }
+
+      def render(node)
+        case node
+
+        when Nodes::Comparison, Nodes::And
+          BINARY_OPERATORS[node.class]
+
+        when Nodes::Date, Nodes::Time
+          node.value.iso8601
+
+        when Nodes::Boolean, Nodes::Number
+          node.value
+
+        when Nodes::Word
+          node.value.to_s
+
+        when Nodes::Grouping::Close
+          ' )'
+
+        when Nodes::Grouping::Open
+          '( '
+
+        else
+          fail ArgumentError.new("Unrecognized node type: #{node.class}")
```

Comment:
I used to be a fan of Seattle-school ruby, but I got a lot of flak, so now I tend to avoid poetry mode.